packages/medusa/src/services/cart.js

```diff
@@ -450,43 +434,22 @@ class CartService extends BaseService {
         })
       }
 
+      // Confirm inventory or throw error
+      await this.inventoryService_
```

Comment:
Sorry, I take my previous comment back. Since the condition for current item would result in a new larger quantity, we need to have the check both places :) Please add.